examples/misc/anchored_artists.py

```diff
@@ -56,48 +69,57 @@ def __init__(self, transform, width, height, angle, loc,
 
 
-class AnchoredDrawingArea(AnchoredOffsetbox):
-    def __init__(self, width, height, xdescent, ydescent,
-                 loc, pad=0.4, borderpad=0.5, prop=None, frameon=True):
-        self.da = DrawingArea(width, height, xdescent, ydescent)
-        super().__init__(loc, pad=pad, borderpad=borderpad,
-                         child=self.da, prop=None, frameon=frameon)
+def draw_ellipse(ax):
+    """
+    Draw an ellipse of width=0.1, height=0.15 in data coordinates
+    """
+    ae = AnchoredEllipse(ax.transData, width=0.1, height=0.15, angle=0.,
+                         loc=3, pad=0.5, borderpad=0.4, frameon=True)
 
+    ax.add_artist(ae)
 
-if __name__ == "__main__":
 
-    import matplotlib.pyplot as plt
+class AnchoredSizeBar(AnchoredOffsetbox):
+    def __init__(self, transform, size, label, loc,
+                 pad=0.1, borderpad=0.1, sep=2, prop=None, frameon=True):
+        """
+        Draw a horizontal bar with the size in data coordinate of the given
+        axes. A label will be drawn underneath (center-aligned).
 
-    ax = plt.gca()
-    ax.set_aspect(1.)
+        pad, borderpad in fraction of the legend font size (or prop)
+        sep in points.
+        """
+        self.size_bar = AuxTransformBox(transform)
+        self.size_bar.add_artist(Rectangle((0, 0), size, 0, ec="black", lw=1.0))
 
-    at = AnchoredText("Figure 1a",
-                      loc=2, frameon=True)
-    at.patch.set_boxstyle("round,pad=0.,rounding_size=0.2")
-    ax.add_artist(at)
+        self.txt_label = TextArea(label, minimumdescent=False)
 
-    from matplotlib.patches import Circle
-    ada = AnchoredDrawingArea(20, 20, 0, 0,
-                              loc=1, pad=0., frameon=False)
-    p = Circle((10, 10), 10)
-    ada.da.add_artist(p)
-    ax.add_artist(ada)
+        self._box = VPacker(children=[self.size_bar, self.txt_label],
+                            align="center",
+                            pad=0, sep=sep)
 
-    # draw an ellipse of width=0.1, height=0.15 in the data coordinate
-    ae = AnchoredEllipse(ax.transData, width=0.1, height=0.15, angle=0.,
-                         loc=3, pad=0.5, borderpad=0.4, frameon=True)
+        super().__init__(loc, pad=pad, borderpad=borderpad,
+                         child=self._box, prop=prop, frameon=frameon)
 
-    ax.add_artist(ae)
 
-    # draw a horizontal bar with length of 0.1 in Data coordinate
-    # (ax.transData) with a label underneath.
+def draw_sizebar(ax):
+    """
+    Draw a horizontal bar with length of 0.1 in data coordinates,
+    with a fixed label underneath.
+    """
     asb = AnchoredSizeBar(ax.transData,
                           0.1,
                           r"1$^{\prime}$",
                           loc=8,
                           pad=0.1, borderpad=0.5, sep=5,
                           frameon=False)
     ax.add_artist(asb)
+ax = plt.gca()
+ax.set_aspect(1.)
+
+draw_text(ax)
+draw_circle(ax)
+draw_ellipse(ax)
+draw_sizebar(ax)
 
-    plt.draw()
```

Comment:
And I got rid of this line. It doesn't do anything useful if you call `show` next.